### ISSUE TITLE
Something about gradient_clip

### ISSUE THREAD

**Issue by DouSong** [twtygqyy/pytorch-vdsr#2]
Hello @twtygqyy thank you for your code, that helped a lot. But there is a  issue I  want to ask you: gradient clip avoid the gradient explode, plays an important role in VDSR(implemented by MatConvNet). But when I reproduce the VDSR in caffe , gradient clip shows little effect(I also try many values). Maybe the different clip implement in caffe,MatConvNet and Torch . So , Could you please tell me your view on this issue? Thanks !!

**Comment by twtygqyy** [comment]
Hi, @DouSong, I'm glad that the code can help. 
About your question in terms of gradient clipping, it is true that the value of the threshold (within a certain range) will not make the result very much different. Since the purpose of applying gradient clipping is to avoid the training from getting gradient exploding. 
If you use a very large value, let's say 100, as threshold for gradient clipping, the gradients will explode in a very early stage during the training and nothing can be learnt after that. On the other hand, if you select a very small value, let's say 0.0001 as threshold, then it will take much longer time for the parameters to be converged to optimum during the training.

**Comment by DouSong** [comment]
Thank you so much @twtygqyy ,you are so cool !

**Comment by twtygqyy** [comment]
@DouSong  You're welcome
